compiler/src/iree/compiler/Dialect/Flow/Transforms/FormDispatchRegions.cpp

```diff
@@ -690,13 +691,153 @@ FailureOr<Flow::WorkloadBuilder> getWorkloadBuilder(OpBuilder &builder,
 
   return result;
 }
+/// Returns the position of given dimension.
+/// For dim=2, map=affine_map<(d0, d1, d2) -> (d0, d2), it returns 1
+static Optional<unsigned> getResultPosition(AffineMap &map, unsigned dim) {
+  unsigned pos = 0;
+  for (AffineExpr result : map.getResults()) {
+    if (result.cast<AffineDimExpr>().getPosition() == dim) return pos;
+    pos++;
+  }
+  return llvm::None;
+}
+
+/// Finds contiguous loop ranges from the given linalg.generic.
+static SmallVector<ReassociationIndices> getContiguousLoops(
```

Comment:
IIUC, this is just looking at the iteration space. That captures a majority of cases but is not complete. For example
```
%0 = linalg.generic {
    indexing_maps = [affine_map<(d0, d1) -> (d1, d0)>, affine_map<(d0, d1) -> (d1, d0)>]
    iterator_types = ["parallel", "parallel"]}
    ins(%1 : tensor<10x20xf32>) outs(%2 : tensor<10x20xf32>) {....} -> tensor<10x20xf32>
```
is not caught by this.
The way to approach this would be to start with indexing maps of one of the operands and see if that sequence of positions appears in every other indexing map, and use the largest such sequence. The underlying transformation used here is capable of folding such cases as well.
I wont block this, but it would be a good thing to do as a follow up to improve the logic to not base things off the position of the iteration loops (which are for the most part an arbitrary ordering and can be changed around freely), but rather use the ordering based on the range of the indexing maps (which is rooted in something more "real", i.e. the tensor layout which IREE assumes is row-major when no encoding is specified).

Reply:
I address this case as well.